src/webrtc/groupCall.ts

```diff
@@ -0,0 +1,559 @@
+import EventEmitter from "events";
+import { CallFeed } from "./callFeed";
+import { MatrixClient } from "../client";
+import { randomString } from "../randomstring";
+import { CallErrorCode, CallState, CallType, MatrixCall } from "./call";
+import { RoomMember } from "../models/room-member";
+import { Room } from "../models/room";
+import { logger } from "../logger";
+import { Callback } from "../client";
+import { ReEmitter } from "../ReEmitter";
+import { GroupCallParticipant, GroupCallParticipantEvent } from "./groupCallParticipant";
+import { SDPStreamMetadataPurpose } from "./callEventTypes";
+
+export enum GroupCallEvent {
+    Entered = "entered",
+    Left = "left",
+    ActiveSpeakerChanged = "active_speaker_changed",
+    ParticipantsChanged = "participants_changed",
+    LocalMuteStateChanged = "local_mute_state_changed",
+}
+
+const CONF_ROOM = "me.robertlong.conf";
+const CONF_PARTICIPANT = "me.robertlong.conf.participant";
+
+export class GroupCall extends EventEmitter {
+    public entered = false;
+    public activeSpeaker: GroupCallParticipant;
+    public localParticipant: GroupCallParticipant;
+    public participants: GroupCallParticipant[] = [];
+    public room: Room;
+    public activeSpeakerSampleCount = 8;
+    public activeSpeakerInterval = 1000;
+    public speakingThreshold = -80;
+    public participantTimeout = 1000 * 15;
+
+    private speakerMap: Map<RoomMember, number[]> = new Map();
+    private presenceLoopTimeout?: number;
+    private activeSpeakerLoopTimeout: number;
+    private reEmitter: ReEmitter;
+
+    constructor(
+        private client: MatrixClient,
+        roomId: string,
+        public type: CallType,
```

Comment:
We are slowly killing of `CallType`s in `MatrixCall` and I think we should do the same here. This feels even more important for conferences - users should be allowed to join in whatever way they want (no camera, no mic, no nothing, whatever :D)

Reply:
We've decided to keep this as the initial state in the MSC but I think the js-doc should reflect this is just the initial type